src/modules/matches/matches.controller.ts

```diff
@@ -34,13 +39,23 @@ export class MatchesController {
     @ApiParam({ name: 'id', type: String })
     @ApiBody({ type: [Match] })
     @Put('/:id')
-    async update(@Param('id') id : string, @Body() match : Match): Promise<any> {
-        return this.matchesService.update(id, match);
+    @HttpCode(204)
+    async update(@Param('id') id: string, @Body() match: Match): Promise<any> {
+        const result = await this.matchesService.update(id, match);
+
+        if (result.affected === 0) {
+            throw new NotFoundException();
+        }
     }
 
     @UseGuards(JwtAuthGuard)
     @Delete('/:id')
-    async delete(@Param('id') id : string ): Promise<any> {
-        return this.matchesService.delete(id);
+    @HttpCode(204)
+    async delete(@Param('id') id: string): Promise<any> {
+        const result = await this.matchesService.delete(id);
+
+        if (result.affected === 0) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏
- Can result be null ? yes  then it could be 'result?.affected'
- why not using a try catch block also here ?

Reply:
matchesService uses TypeOrm delete function which return a Promise<DeleteResult>. As long as I can say, it rejects the promise when no criteria is provided and always resolve the promise otherwise with a DeleteResult.